examples/custom-state-storage/mongo-state-storage.js

```diff
@@ -0,0 +1,62 @@
+const mongodb = require('mongodb');
+const MongoClient = mongodb.MongoClient;
+const Bluebird = require('bluebird');
+
+Bluebird.promisifyAll(MongoClient);
+
+class MongoDbStore {
+    constructor() {
+        this.url = 'mongodb://localhost/test';
+    }
+
+    connect() {
+        return MongoClient.connect(this.url)
+            .then(client => {
+                return client.db()
+            })
+    }
+
+    load(fn) {
+        return this.connect()
+            .then(db => db.collection('db_migrations').find().toArray())
+            .then(data => {
+                if (!data.length) return fn(null, {});
+                const store = data[0];
+                if (!MongoDbStore.hasProperty(store, 'lastRun') ||
+                    !MongoDbStore.hasProperty(store, 'migrations')) {
+                    return fn(new Error('I`nvalid store file'))
+                }
+                return fn(null, store)
+            }).catch(fn)
+    }
+
+    static hasProperty(store, property) {
+        return Object
+            .prototype
+            .hasOwnProperty
+            .call(store, property);
+    }
+
+    save(set, fn) {
+        return this.connect()
+            .then(db => db.collection('db_migrations')
+                .update({},
+                    {
```

Comment:
Sorry for all the "make it more simple" comments, but again this can be much more simple if you just do one upsert which is just the whole set as json.